### ISSUE TITLE
RFC: Lazy PageInfo resolving

### ISSUE THREAD

**Issue by mxstbr** [StellateHQ/fuse#155]
## Summary

When using `t.connection`, I **have to always resolve `hasNextPage` and `hasPreviousPage`** in `resolve()`, even if either of those fields isn't requested. Example: https://github.com/rauchg/next-ai-news/pull/8/files#diff-b6d759f387b5e305d570ddfbb9d727c6c8e01863944286e7da41d36db5bdae0cR49-R87

## Proposed Solution

Some way to define individual resolvers for `hasNextPage` and `hasPreviousPage` that only get run if the corresponding fields are actually in the query.

**Comment by lkuoch** [comment]
Definitely! I believe this could also be more efficient as well with querying the db only once. This could be done by returning a tuple of the count of items based on the offset as well as the query result.
